packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -67,5 +209,61 @@ protected override MethodProvider[] BuildMethods()
 
             return methods.ToArray();
         }
+
+        private ParameterProvider[] GetPrimaryConstructorParameters()
+        {
+            if (_apiKeyAuthField != null)
+            {
+                return [_endpointParameter, _apiKeyAuthField.AsParameter, _clientOptionsParameter];
+            }
+
+            return [_endpointParameter, _clientOptionsParameter];
+        }
+
+        private ParameterProvider BuildClientEndpointParameter()
+        {
+            for (var i = 0; i < _inputClient.Parameters.Count; i++)
+            {
+                var inputClientParam = _inputClient.Parameters[i];
+                if (inputClientParam.IsEndpoint)
+                {
+                    var parameterProvider = new ParameterProvider(inputClientParam);
+                    var knownEndpointParam = KnownParameters.Endpoint;
+                    var description = inputClientParam.Description ?? $"{knownEndpointParam.Description}";
+                    var initializationValue = GetParameterInitializationValue(inputClientParam);
+                    var endpointValue = initializationValue != null ? New.Instance(knownEndpointParam.Type, initializationValue) : null;
+
+                    return new(
+                        knownEndpointParam.Name,
+                        $"{description}",
+                        knownEndpointParam.Type,
+                        initializationValue: endpointValue)
+                    {
+                        Validation = parameterProvider.Validation
+                    };
+                }
+            }
+
+            return KnownParameters.Endpoint;
+        }
+
+        private static ValueExpression? GetParameterInitializationValue(InputParameter inputParameter)
```

Comment:
At least in naming it seems we are still mixing up default and initialization expressions.

Also not sure we want to use System.Convert ever, do we have an actual use case for this?

Reply:
Now that this should only ever be called for the endpoint parameter, I don't think we need to use System.Convert. I have removed it.